contracts/metatx/MinimalForwarder.sol

```diff
@@ -23,59 +28,85 @@ contract MinimalForwarder is EIP712 {
         uint256 value;
         uint256 gas;
         uint256 nonce;
+        uint48 deadline;
         bytes data;
     }
 
-    bytes32 private constant _TYPEHASH =
-        keccak256("ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)");
-
-    mapping(address => uint256) private _nonces;
+    bytes32 private constant _FORWARD_REQUEST_TYPEHASH =
+        keccak256(
+            "ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,uint48 deadline,bytes data)"
+        );
 
     /**
      * @dev The request `from` doesn't match with the recovered `signer`.
      */
     error MinimalForwarderInvalidSigner(address signer, address from);
 
     /**
-     * @dev The request nonce doesn't match with the `current` nonce for the request signer.
+     * @dev The request `value` doesn't match with the `msg.value`.
+     */
+    error MinimalForwaderMismatchedValue(uint256 msgValue, uint256 value);
+
+    /**
+     * @dev The list of requests length doesn't match with the list of signatures length.
      */
-    error MinimalForwarderInvalidNonce(address signer, uint256 current);
+    error MinimalForwaderInvalidBatchLength(uint256 requestsLength, uint256 signaturesLength);
```

Comment:
This error is not used at the moment.

Are we planning to add batching?

Reply:
> Are we planning to add batching?

Did we decide not to do so? I think we agreed a while ago.

Batching added.